src/ndpp_init.F90

```diff
@@ -0,0 +1,604 @@
+module ndpp_initialize
+
+  use ace_header,   only: Nuclide, XsListing
+  use constants
+  use dict_header,  only: DictCharInt, DICT_NULL
+  use error,        only: fatal_error, warning
+  use global
+  use ndpp_header,  only: Ndpp
+  use ndpp_ops,     only: ndpp_read
+  use output,       only: write_message
+  use search
+  use string,       only: ends_with, to_lower, starts_with, to_str
+  use xml_interface
+
+  implicit none
+
+  ! ndpp_lib.xml preprocessed data listings and associated data.
+  type(XsListing), allocatable, target, save :: ndpp_listings(:)
+  type(DictCharInt), save                    :: ndpp_listing_dict
+
+contains
+
+!===============================================================================
+! READ_NDPP_LIB reads all the cross sections for the problem and stores them in
+! nuclides and sab_tables arrays
+!===============================================================================
+
+  subroutine read_ndpp_data()
+    type(Nuclide), pointer    :: nuc => null() ! Current working nuclide
+    type(SAlphaBeta), pointer :: sab => null() ! Current working SAB table
+    type(XsListing), pointer  :: ndpp_listing => null()
```

Comment:
In sticking with recent changes, locally-declared pointers should not be initialized to null() at declaration since this causes them to have static duration.